lib/tasks/solution_path_cleanup.rake

```diff
@@ -0,0 +1,17 @@
+task :environment do
+  require File.expand_path('../../app',
+                           File.dirname(__FILE__)) # your Sinatra app
+end
+
+namespace :db do
+  desc "Make all solution paths relative to the problem directory, not root"
+  task solution_path_cleanup: :environment do
+    Submission.all.each do |submission|
+      new_filename = Code.new(submission.filename)
+      puts "Updating #{submission.key}"
+      submission.update_attributes!(filename: new_filename)
```

Comment:
If this raises an error, let's print the key that failed to STDERR and then keep going. Then I'll use that to investigate any errors, but it will likely have fixed the majority of things.